### PR TITLE
Italy provinces v2

### LINKED ISSUE #131: Revisit Italy provinces
South Sardinia got the ISO code so [I added it](https://www.wikidata.org/w/index.php?title=Q23498165&diff=prev&oldid=1037002012), but the old Sicily provinces are still listed under the Region so the I had to tweak the [previous query](https://github.com/elastic/ems-file-service/blob/master/sources/it/provinces.hjson#L52-L111) to return [all the 107 boundaries](https://sophox.org/#%20%20%20%20%23defaultView%3AMapRegions%0A%20%20%20%20SELECT%20%3Fid%20%3Fiso_3166_2%20%3Fistat%20%3Flabel_en%20%3Flabel_it%0A%20%20%20%20WHERE%20%7B%0A%20%20%20%20%20%20SERVICE%20%3Chttps%3A%2F%2Fquery.wikidata.org%2Fsparql%3E%20%7B%0A%20%20%20%20%20%20%20%20%7B%0A%20%20%20%20%20%20%20%20%20%20SELECT%20%0A%20%20%20%20%20%20%20%20%20%20%20%20%3Fid%0A%20%20%20%20%20%20%20%20%20%20%20%20%28SAMPLE%28%3Fid2%29%20AS%20%3Fid2%29%20%0A%20%20%20%20%20%20%20%20%20%20%20%20%28SAMPLE%28%3Fiso_3166_2%29%20AS%20%3Fiso_3166_2%29%20%0A%20%20%20%20%20%20%20%20%20%20%20%20%28SAMPLE%28%3Fistat%29%20AS%20%3Fistat%29%20%0A%20%20%20%20%20%20%20%20%20%20%20%20%28SAMPLE%28REPLACE%28%3Flabel_en%2C%20%22Province%20of%20%22%2C%20%22%22%29%29%20as%20%3Flabel_en%29%0A%20%20%20%20%20%20%20%20%20%20%20%20%28SAMPLE%28REPLACE%28%3Flabel_it%2C%20%22provincia%20di%20%22%2C%20%22%22%29%29%20as%20%3Flabel_it%29%0A%20%20%20%20%20%20%20%20%20%20%20%20WHERE%20%7B%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20VALUES%20%28%3Fentity%29%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%28wd%3AQ38%29%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20%3Fentity%20%28wdt%3AP150%2Fwdt%3AP150%29%20%3Fid.%0A%20%20%20%20%20%20%20%20%20%20%20%20%3Fid%20wdt%3AP300%20%3Fiso_3166_2.%0A%20%20%20%20%20%20%20%20%20%20%20%20MINUS%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3Fid%20%28wdt%3AP576%7Cwdt%3AP582%29%20%3Fend.%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20FILTER%28%3Fid%20NOT%20IN%28%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20wd%3AQ16112%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20wd%3AQ16150%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20wd%3AQ16154%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20wd%3AQ16166%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20wd%3AQ16208%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20wd%3AQ16227%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20wd%3AQ16251%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20wd%3AQ16277%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20wd%3AQ16288%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20wd%3AQ16184%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20wd%3AQ16247%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20wd%3AQ16293%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20wd%3AQ16294%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%29%29%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20OPTIONAL%20%7B%20%3Fid%20wdt%3AP635%20%3Fistat.%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20OPTIONAL%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3Fid%20rdfs%3Alabel%20%3Flabel_en.%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20FILTER%28%28LANG%28%3Flabel_en%29%29%20%3D%20%22en%22%29%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20OPTIONAL%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3Fid%20rdfs%3Alabel%20%3Flabel_it.%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20FILTER%28%28LANG%28%3Flabel_it%29%29%20%3D%20%22it%22%29%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%7D%20UNION%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%23%20ISO%20Code%20IT-AO%20refers%20to%20a%20province%20that%20no%20longer%20exists.%20The%20GeoLite2%20IP%20database%20lists%20cities%20from%20the%20Aosta%20Valley%0A%20%20%20%20%20%20%20%20%20%20%20%20%23%20regiohn%20so%20we%27ll%20use%20the%20region%20boundaries%20for%20the%20IT-AO%20code.%0A%20%20%20%20%20%20%20%20%20%20%20%20VALUES%20%3Fentity%20%7B%20wd%3AQ38%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20%3Fentity%20wdt%3AP150%20%3Fid%20.%0A%20%20%20%20%20%20%20%20%20%20%20%20FILTER%20%28%20%3Fid%20%3D%20wd%3AQ1222%20%29%0A%20%20%20%20%20%20%20%20%20%20%20%20VALUES%20%3Fiso_3166_2%20%7B%20%22IT-AO%22%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20OPTIONAL%20%7B%20%3Fid%20wdt%3AP635%20%3Fistat.%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20OPTIONAL%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3Fid%20rdfs%3Alabel%20%3Flabel_en.%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20FILTER%28%28LANG%28%3Flabel_en%29%29%20%3D%20%22en%22%29%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20OPTIONAL%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3Fid%20rdfs%3Alabel%20%3Flabel_it.%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20FILTER%28%28LANG%28%3Flabel_it%29%29%20%3D%20%22it%22%29%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%7D%20UNION%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%23%20Sicily%20comunes%0A%20%20%20%20%20%20%20%20%20%20%20%20%3Fid%20wdt%3AP31%2Fwdt%3AP279%2a%20%20wd%3AQ21190155.%0A%20%20%20%20%20%20%20%20%20%20%20%20%3Fid%20wdt%3AP300%20%3Fiso_3166_2.%0A%20%20%20%20%20%20%20%20%20%20%20%20%0A%20%20%20%20%20%20%20%20%20%20%20%20OPTIONAL%20%7B%20%3Fid%20wdt%3AP635%20%3Fistat.%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20OPTIONAL%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3Fid%20rdfs%3Alabel%20%3Flabel_en.%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20FILTER%28%28LANG%28%3Flabel_en%29%29%20%3D%20%22en%22%29%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20OPTIONAL%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3Fid%20rdfs%3Alabel%20%3Flabel_it.%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20FILTER%28%28LANG%28%3Flabel_it%29%29%20%3D%20%22it%22%29%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%7D%20UNION%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%23%20Sicily%20metropolitan%20cities%0A%20%20%20%20%20%20%20%20%20%20%20%20VALUES%20%28%3Fid%29%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%28wd%3AQ20795016%29%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%28wd%3AQ20991250%29%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%28wd%3AQ20991246%29%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20%3Fid%20wdt%3AP300%20%3Fiso_3166_2.%0A%20%20%20%20%20%20%20%20%20%20%20%20%0A%20%20%20%20%20%20%20%20%20%20%20%20OPTIONAL%20%7B%20%3Fid%20wdt%3AP635%20%3Fistat.%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20OPTIONAL%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3Fid%20rdfs%3Alabel%20%3Flabel_en.%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20FILTER%28%28LANG%28%3Flabel_en%29%29%20%3D%20%22en%22%29%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20OPTIONAL%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3Fid%20rdfs%3Alabel%20%3Flabel_it.%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20FILTER%28%28LANG%28%3Flabel_it%29%29%20%3D%20%22it%22%29%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20GROUP%20BY%20%3Fid%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%0A)

```sparql
     #defaultView:MapRegions
    SELECT ?id ?iso_3166_2 ?istat ?label_en ?label_it
    WHERE {
      SERVICE <https://query.wikidata.org/sparql> {
        {
          SELECT 
            ?id
            (SAMPLE(?id2) AS ?id2) 
            (SAMPLE(?iso_3166_2) AS ?iso_3166_2) 
            (SAMPLE(?istat) AS ?istat) 
            (SAMPLE(REPLACE(?label_en, "Province of ", "")) as ?label_en)
            (SAMPLE(REPLACE(?label_it, "provincia di ", "")) as ?label_it)
            WHERE { {
            VALUES (?entity) {
              (wd:Q38)
            }
            ?entity (wdt:P150/wdt:P150) ?id.
            ?id wdt:P300 ?iso_3166_2.
            MINUS {
              ?id (wdt:P576|wdt:P582) ?end.
              FILTER(?id NOT IN(
              wd:Q16112,
              wd:Q16150,
              wd:Q16154,
              wd:Q16166,
              wd:Q16208,
              wd:Q16227,
              wd:Q16251,
              wd:Q16277,
              wd:Q16288,
              wd:Q16184,
              wd:Q16247,
              wd:Q16293,
              wd:Q16294
              ))
            }
            OPTIONAL { ?id wdt:P635 ?istat. }
            OPTIONAL {
              ?id rdfs:label ?label_en.
              FILTER((LANG(?label_en)) = "en")
            }
            OPTIONAL {
              ?id rdfs:label ?label_it.
              FILTER((LANG(?label_it)) = "it")
            }
          } UNION {
            # ISO Code IT-AO refers to a province that no longer exists. The GeoLite2 IP database lists cities from the Aosta Valley
            # regiohn so we'll use the region boundaries for the IT-AO code.
            VALUES ?entity { wd:Q38 }
            ?entity wdt:P150 ?id .
            FILTER ( ?id = wd:Q1222 )
            VALUES ?iso_3166_2 { "IT-AO" }
            OPTIONAL { ?id wdt:P635 ?istat. }
            OPTIONAL {
              ?id rdfs:label ?label_en.
              FILTER((LANG(?label_en)) = "en")
            }
            OPTIONAL {
              ?id rdfs:label ?label_it.
              FILTER((LANG(?label_it)) = "it")
            }
          } UNION {
            # Sicily comunes
            ?id wdt:P31/wdt:P279*  wd:Q21190155.
            ?id wdt:P300 ?iso_3166_2.
            
            OPTIONAL { ?id wdt:P635 ?istat. }
            OPTIONAL {
              ?id rdfs:label ?label_en.
              FILTER((LANG(?label_en)) = "en")
            }
            OPTIONAL {
              ?id rdfs:label ?label_it.
              FILTER((LANG(?label_it)) = "it")
            }
          } UNION {
            # Sicily metropolitan cities
            VALUES (?id) {
              (wd:Q20795016)
              (wd:Q20991250)
              (wd:Q20991246)
            }
            ?id wdt:P300 ?iso_3166_2.
            
            OPTIONAL { ?id wdt:P635 ?istat. }
            OPTIONAL {
              ?id rdfs:label ?label_en.
              FILTER((LANG(?label_en)) = "en")
            }
            OPTIONAL {
              ?id rdfs:label ?label_it.
              FILTER((LANG(?label_it)) = "it")
            }
          }
          }
          GROUP BY ?id
        }
      }
    }
```

### PR DESCRIPTION
**Pull request by jsanz**
Closes #131

### CODE REVIEW

File: sources/it/provinces_v2.hjson [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    WHERE {
+      SERVICE <https://query.wikidata.org/sparql> {
+        {
+          SELECT
+            ?id
+            (SAMPLE(?id2) AS ?id2)
+            (SAMPLE(?iso_3166_2) AS ?iso_3166_2)
+            (SAMPLE(?istat) AS ?istat)
+            (SAMPLE(REPLACE(?label_en, "Province of ", "")) as ?label_en)
+            (SAMPLE(REPLACE(?label_it, "provincia di ", "")) as ?label_it)
```
Comment (nickpeihl) [inline sources/it/provinces_v2.hjson]: Is it possible to do multiple `REPLACE...` functions for each field? Maybe we can drop `Libero consorzio comunale di` from the communes.

**Comment by nickpeihl** [comment]
The revised "istat" ids are the significant breaking change in this PR. Hopefully name changes will be less of an issue if we drop support for joins on `type: property` fields in future versions of Kibana.

File: sources/it/provinces_v2.hjson [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    tr: İtalya'nın illeri
+    uk: провінція Італії
+    ur: اٹلی کے صوبے
+    vec: Provincia itałiana
+    vls: Italioansche provinsjes
+    zh: 意大利各省
+  }
+  emsFormats: [{
+    type: geojson
+    file: italy_provinces_v2.geo.json
```
[the remark this answers is not in the document]
Comment (jsanz) [reply sources/it/provinces_v2.hjson]: thanks, I didn't have that clear, sorry.

File: sources/it/provinces_v2.hjson (hunk 1 above)
Comment (jsanz) [reply sources/it/provinces_v2.hjson]: yep, it's possible it actually accepts a regex so not super beautiful but this made it

```sparql
(SAMPLE(REPLACE(?label_en, "(Province of |Metropolitan City of |Libero consorzio comunale di )", "")) as ?label_en)
(SAMPLE(REPLACE(?label_it, "((P|p)rovincia( autonoma|) (di |del |dell'|della )|città metropolitana di |Libero consorzio comunale di )", "")) as ?label_it)
```

**Pull request merged**